storage/merge.go

```diff
@@ -393,12 +402,14 @@ func (c *genericMergeSeriesSet) Next() bool {
 
 func (c *genericMergeSeriesSet) At() Labels {
 	if len(c.currentSets) == 1 {
+		c.mergedSeries++
```

Comment:
This seems to assume that `At()` is only called once per `Next()` ?
I don't think that is guaranteed.

Reply:
Thanks for catching this. I'll move this into `Next()`.